internal/util/pkgutil/pkgutil.go

```diff
@@ -51,22 +93,65 @@ func FindLocalSubpackages(path string) ([]string, error) {
 			return nil
 		}
 
-		pkgPath := filepath.Dir(p)
-		localSubPkgs = append(localSubPkgs, pkgPath)
+		if p == filepath.Join(path, kptfilev1alpha2.KptFileName) {
+			return nil
+		}
 
+		pkgPath := filepath.Dir(p)
 		kf, err := kptfileutil.ReadFile(pkgPath)
 		if err != nil {
 			return err
 		}
-		for _, sp := range kf.Subpackages {
-			spPath := filepath.Join(pkgPath, sp.LocalDir)
-			remoteSubPkgs = append(remoteSubPkgs, spPath)
+
+		if kf.Upstream == nil {
+			localSubPkgs = append(localSubPkgs, pkgPath)
+		} else {
+			remoteSubPkgs = append(remoteSubPkgs, pkgPath)
 		}
 		return nil
 	})
 	return localSubPkgs, err
 }
 
+// FindRemoteDirectSubpackages
```

Comment:
Please explain what this is doing.

Reply:
Added a full description of this function.